File: SimpleLinearRegression_0308.py
#Simple Linear Regression
"""
Created on Friday 03/08/2019 11:25 PM
@author: Deepak Pradeep Kumar
"""
#Importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#Importing the data
dataset = pd.read_csv('Salary.csv')

# Validate the dataset
dataset.describe()

# Split into Dependent and Independent variables
X = dataset.iloc[:,:-1].values
Y = dataset.iloc[:, 1].values

#X = dataset.YearsOfExperience
#Y = dataset.Salary

# No Manipulation or Imputation of data is needed as this is a Sample and clean dataset.
#Splitting the dataset to Train and Test data.
from sklearn.cross_validation import train_test_split
Xtrain, Xtest, Ytrain, Ytest = train_test_split(X,Y, test_size = 0.3, random_state = 1)

#Fitting the Simple Linear Regression Model
from sklearn.linear_model import LinearRegression
LinearRegressor = LinearRegression()
LinearRegressor.fit(Xtrain, Ytrain)

# Predicting the Test Set Results
YPredictor = LinearRegressor.predict(Xtest)

# Evaluating the Intercept
LinearRegressor.intercept_

#Plotting the Graph for Predicted Test Set Results
plt.scatter(Xtest, Ytest, color = 'blue')
plt.plot(Xtest, YPredictor, color = 'red')
plt.title('Experience vs Salary')
plt.xlabel('Number of Years of Experience')
plt.ylabel('Salary')
plt.show()

YPred = LinearRegressor.predict(16.5)

File: SimpleLinearRegression_0308.R

# Simple Linear Regression File
# Author : Deepak Pradeep Kumar
setwd("D:\\Personal\\DataPlatformGeeks\\Mar092019")

#Import the dataset
dataset = read.csv("Salary.csv")

#dividing the dataset into trainingset and testset
#install.packages('caTools')
library(caTools)
set.seed(123)

fragment = sample.split(dataset$Salary, SplitRatio = 2/3, group = NULL)
trainingset = subset(dataset, fragment == TRUE)
testset = subset(dataset, fragment == FALSE)

#Fitting simple Linear model to training set
LinearRegressor = lm(formula = Salary ~ YearsOfExperience,
                     data = trainingset )

summary(LinearRegressor)

#Predicting the Test Set Results
Predictor = predict(LinearRegressor, newdata = testset)

# Plotting the graph on testset results using ggplot2
#install.packages('ggplot2')
library(ggplot2)

ggplot() + 
  geom_point(aes(x = testset$YearsOfExperience, y = testset$Salary), 
             colour = 'blue') + 
  geom_line(aes(x = testset$YearsOfExperience, y= Predictor),
            colour = 'red') +
  ggtitle('Salary vs Experience') +
  xlab('YearsOfExperience') +
  ylab('Salary')

# Checking and validating the predicted results

testset1 = read.csv('validating.csv')

predict(LinearRegressor, newdata = testset1)










